### PR TITLE
TLS/HTTPS (semi) support

### PR DESCRIPTION
**Pull request by dobby5**
See: https://github.com/getnamo/socketio-client-ue4/issues/39#issuecomment-846415233

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,8 +22,31 @@ Current platform issues:
 * Xbox/PS4 platform untested - see [issue 117](https://github.com/getnamo/socketio-client-ue4/issues/117)
 * Lumin platform untested - see [issue 114](https://github.com/getnamo/socketio-client-ue4/issues/114)
 
-HTTPS currently not yet supported
-* OpenSSL Support - [issue39](https://github.com/getnamo/socketio-client-ue4/issues/39), temporary [workaround available](https://github.com/getnamo/socketio-client-ue4/issues/72#issuecomment-371956821).
+HTTPS is only supported through a little change in the source code.
+* OpenSSL Support - <a href="https://github.com/dobby5/socketio-client-ue4/blob/0d3420f626704802d457e125bf01d0d4d26bbfe3/Source/SocketIOLib/SocketIOLib.Build.cs#L63-L73">Remove this declaration and compile the code, that's all.</a>
```
Comment (dobby5) [inline README.md:26]: If there is a merge please change this line with your branch @getnamo

**Comment by getnamo** [comment]
I'm not a fan of overriding SSL support by dropping non-ssl as that mode is far more common in use case. What can be done is to merge this into a ssl branch until we figure out a way to swap support via url lookup (https -> ssl, http-> standard)

**Comment by getnamo** [comment]
added merge to secondary branch here: https://github.com/getnamo/socketio-client-ue4/tree/ssl

**Comment by getnamo** [comment]
open future pull requests there until we manage to support both methods and it becomes ready for master

**Comment by getnamo** [comment]
I think I might have an idea on how to get the url based switch support. Will try to find a day or two to explore it

**Comment by dobby5** [comment]
I thing it is possible to including a variable at the said places, but I see the problem mainly at the place https://github.com/getnamo/socketio-client-ue4/blob/d0cad0135c1aca1ea4d891697c21f96767741fc0/Source/SocketIOLib/Private/internal/sio_client_impl.h#L38-L62, in the inclusion of the SSL or non-SSL hpp data.

**Comment by dobby5** [comment]
I thing it is possible to including a variable at the said places, but I see the problem mainly at the place https://github.com/getnamo/socketio-client-ue4/blob/d0cad0135c1aca1ea4d891697c21f96767741fc0/Source/SocketIOLib/Private/internal/sio_client_impl.h#L38-L62, in the inclusion of the SSL or non-SSL hpp data.

**Comment by getnamo** [comment]
Let's see that's used in https://github.com/getnamo/socketio-client-ue4/blob/ssl/Source/SocketIOLib/Private/internal/sio_client_impl.h#L81 which is used in https://github.com/getnamo/socketio-client-ue4/blob/ssl/Source/SocketIOLib/Private/internal/sio_client_impl.cpp#L283 and https://github.com/getnamo/socketio-client-ue4/blob/ssl/Source/SocketIOLib/Private/internal/sio_client_impl.cpp#L470 and https://github.com/getnamo/socketio-client-ue4/blob/ssl/Source/SocketIOLib/Private/internal/sio_client_impl.cpp#L514.

We might be able to keep two type of configs and pass them in depending on connection types I think. The above is only a typedef so I do think we can keep two types

**Comment by getnamo** [comment]
Yeah we'd need a different handler for on_message in tls variant https://github.com/getnamo/socketio-client-ue4/blob/ssl/Source/SocketIOLib/Private/internal/sio_client_impl.cpp#L76 with the tls config

**Comment by dobby5** [comment]
Good morning, if I understand it correctly, your goal is basically to make these changes, right?

https://github.com/socketio/socket.io-client-cpp/pull/137

**Comment by dobby5** [comment]
I have taken the changes from joelnordell and adapted them to the new version of our socket.io-client-cpp.

I still get an error with the "class/struct" maybe you have an idea how we can fix this.

I have pushed the changes into a new branch:
https://github.com/dobby5/socketio-client-ue4/tree/tls_testing

**Comment by getnamo** [comment]
Going to work on it within https://github.com/getnamo/socketio-client-ue4/pull/267

At a first glance, I'm not fully convinced at the templated approach joelnordell did, seems unnecessarily complicated, but if it works, it could be a quick enough patch. Will know more later today.

**Comment by getnamo** [comment]
Got it to compile and pulled into https://github.com/getnamo/socketio-client-ue4/commits/ssl-dev

Emit still works, but receiving events broke, TBC

**Comment by dobby5** [comment]
Hello @getnamo,

I hope that we can find a quick fix for this. I will have a look at your changes during the day. Maybe you will find another approach to solution. I am also curious how you will solve it.

**Comment by dobby5** [comment]
@getnamo Do you have any news? Good and fast work by the way with the UE5 update.

**Comment by getnamo** [comment]
Still pushing the SSL refactor for next week (it's a lot more involved than the ue5 upgrade) as I'm out of time for sio this week. Will know more then.

**Comment by dobby5** [comment]
Okay good

**Pull request closed without merging**